src/nvim/tui/tui.c

```diff
@@ -69,6 +70,9 @@ typedef struct {
   bool can_use_terminal_scroll;
   bool mouse_enabled;
   bool busy;
+  /* use int instead ? */
+  // cursorentry_T cursor_shapes[SHAPE_IDX_COUNT]; /**!< is it fine to reuse cursorentry_T here ? */
+  int cursor_shapes[SHAPE_IDX_COUNT]; /**!< is it fine to reuse cursorentry_T here ? */
```

Comment:
As I see it, there are some members in cursorentry_T that don't make sense for the TUI (mshape/blinkoff ?). So I thought I could save a few bytes by generating directly in tui.c, upon cursor_shapes receptions the control sequence and discard the useless stuff. But if you intend to support blinkon/blinkoff in the TUI with timers, it makes sense to regenerate the cursorentry_T.